src/backend/cuda/nvrtc/cache.cpp

```diff
@@ -149,6 +157,107 @@ void Kernel::getScalar(T &out, const char *name) {
 template void Kernel::setScalar<int>(const char *, int);
 template void Kernel::getScalar<int>(int &, const char *);
 
+#if !defined(OS_WIN)
+string getHomeDirectory() {
+    string home = getEnvVar("XDG_CONFIG_HOME");
+    if (!home.empty()) return home;
+
+    home = getEnvVar("HOME");
+    if (!home.empty()) return home;
+
+    home = getpwuid(getuid())->pw_dir;
+}
+#else
+string getTemporaryDirectory() {
+    DWORD bufSize = 261;  // limit according to GetTempPath documentation
```

Comment:
We can do what I have mentioned [here](https://github.com/arrayfire/arrayfire/issues/2503#issuecomment-488546754) for windows. Cache will be persistent for at least a given session or even reboot depending on settings.

Reply:
On my machine ``GetTempPath`` returned ``AppData\Local\Temp``, which I think is the default unless the user has overridden the ``TMP`` or ``TEMP`` environment variables. 

``FOLDERID_AppDataProgramData``, which you suggest as an alternative, looks like it was introduced in Windows 10, I don't know if this is an issue for supporting older Windows OSes.

Reply:
That is unfortunate. There must be a way to handle it for older windows versions too I think.

Reply:
``FOLDERID_AppDataProgramData`` is nothing more than ``%LOCALAPPDATA%\ProgramData`` by default, although the Windows docs say:

> This FOLDERID is used internally by .NET applications to enable cross-platform app functionality. It is not intended to be used directly from an application.

We could instead just ask for ``FOLDERID_LocalAppData`` (which should just be ``%LOCALAPPDATA%``), and build the rest ourselves.

Reply:
I will check about windows and your suggestions, will keep you posted.